rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -885,14 +885,14 @@ stream {
         {{ $server.ServerSnippet }}
         {{ end }}
 
-        {{ $usesRewrite := atLeastOneNeedsRewrite $server.Locations }}
+        {{ $usesRegex := atLeastOneUsesRegex $server.Locations }}
         {{ range $location := $server.Locations }}
-        {{ $path := buildLocation $location $usesRewrite }}
+        {{ $path := buildLocation $location $usesRegex }}
         {{ $proxySetHeader := proxySetHeader $location }}
         {{ $authPath := buildAuthLocation $location }}
 
         {{ if not (empty $location.Rewrite.AppRoot)}}
-        if ($uri = /) {
+        if ($uri = / or $uri = "~* ^/") {
```

Comment:
What do you think about introducing a helper template function to check this? Something like `isRootURI(uri string)`